requirements.txt

```diff
@@ -6,3 +6,4 @@ scikit-learn>=0.20.0
 setuptools>=40.1.0
 h5py<3.3
 dataclasses; python_version < '3.7'
+retworkx>=0.10.1
```

Comment:
While I see that `retworkx` dependent has been added the new code also seems to now depend on `matplotlib`. In other areas of Qiskit matplotlib is treated more optionally such that its existence is checked and the code will still work although calls to do things like draw, that a user may choose not to do, which depend on it, will raise an exception informing the user they should install it. So it needs to be a dependent added here or treated as an optional and handled accordingly.

Reply:
@ikkoham @k-tamuraphys This needs to be addressed

Reply:
Thanks for the comment. It has been modified to use `HAS_MATPLOTLIB` to check for the existence of `matplotlib` when `draw`is used. So `matplotlib` is now treated as an option.

Reply:
@woodsp-ibm If this is fine for you can we please resolve this issue?

Reply:
Having matpltlib as optional seems fine. @manoelmarques I doubt we have CI setup for this where it runs without the optional library, i.e. matplotlib, to ensure it works without and continues to do so.